fuel-storage/src/impls.rs

```diff
@@ -35,6 +38,38 @@ impl<'a, T: StorageMutate<Type> + ?Sized, Type: Mappable> StorageMutate<Type> fo
     }
 }
 
+impl<'a, T: StorageSize<Type> + ?Sized, Type: Mappable> StorageSize<Type> for &'a T {
+    fn size_of_value(&self, key: &<Type as Mappable>::Key) -> Result<Option<usize>, Self::Error> {
+        <T as StorageSize<Type>>::size_of_value(self, key)
+    }
+}
+
+impl<'a, T: StorageSize<Type> + ?Sized, Type: Mappable> StorageSize<Type> for &'a mut T {
```

Comment:
```suggestion
impl<'a, T: StorageSize<Type> + ?Sized, Type: Mappable> StorageSize<Type> for &'a T {
```

this doesn't seem like it should require a `mut`